test/test_images/committed-offset/admin.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/Shopify/sarama"
+	"github.com/kelseyhightower/envconfig"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/kafka"
+	kafkatest "knative.dev/eventing-kafka-broker/test/pkg/kafka"
+)
+
+func main() {
+
+	adminConfig := sarama.NewConfig()
+	adminConfig.Version = sarama.V2_0_0_0
+
+	envConfig := &kafkatest.AdminConfig{}
+
+	if err := envconfig.Process("", envConfig); err != nil {
+		log.Fatal("Failed to process environment variables", err)
+	}
+
+	j, _ := json.MarshalIndent(envConfig, "", " ")
+	log.Println(string(j))
+
+	client, err := sarama.NewClient([]string{envConfig.BootstrapServers}, adminConfig)
+	if err != nil {
+		log.Fatal("Failed to create sarama client", err)
+	}
+	consumerGroupLagProvider := kafka.NewConsumerGroupLagProvider(client, sarama.NewClusterAdminFromClient, sarama.OffsetOldest)
+	defer consumerGroupLagProvider.Close()
+
+	lag, err := consumerGroupLagProvider.GetLag(envConfig.Topic, envConfig.Group)
+	if err != nil {
+		log.Fatal("Failed to get lag", err)
+	}
+
+	if lag.Total() != 0 {
```

Comment:
can you print the lag data structure as json, parse it in the test runner and then assert there? I prefer to use these test images only as "data collectors", so we can more flexibly perform assertions and debug in the test runner.

Reply:
Sounds interesting but parsing logs seems less robust.

I'm not sure how we use k8s' events for this kind of communication but maybe we can use a similar pattern?

Reply:
> I'm not sure how we use k8s' events for this kind of communication but maybe we can use a similar pattern?

Way too complicated, I suggest we avoid going down that road. WDYM by parsing logs is less robust?

Reply:
Something like
```
	log.Printf(`{"lag": %d}`, lag.Total())
```

and then in the test should I try to parse every log line looking for the above Json? 

Is this what you mean? It seems weird. :thinking:

Reply:
How about just parse the last log line? Is that reasonable?